File: classes/LLMs/examples/pdf_classification/code/python/classify_papers.py
def classify_papers(processed_papers, llm):
    results = []
    
    for paper in processed_papers:
        paper_results = {
            'filename': paper['filename'],
            'classifications': []
        }
        
        # Process each chunk of the paper
        for chunk in paper['content']:
            response = llm.classify(
                prompt_template.format(text_chunk=chunk.page_content)
            )
            paper_results['classifications'].append(response)
        
        # Aggregate results for the paper
        final_classification = aggregate_classifications(
            paper_results['classifications']
        )
        results.append(final_classification)
    
    return results

File: classes/LLMs/examples/pdf_classification/code/R/classify_papers.R
classify_papers <- function(papers_df, 
                            api_key) {

  # Add progress bar
  total_chunks <- nrow(papers_df)
  pb <- txtProgressBar(min = 0, 
                       max = total_chunks, 
                       style = 3)
  
  # Classify each chunk with progress tracking
  results_df <- papers_df %>%
    mutate(
      classification = map_chr(seq_len(n()), 
                               function(i) {
                                 setTxtProgressBar(pb, i)
                                 result <- classify_text(content[i], 
                                                         api_key)
                                 Sys.sleep(10)  # Add delay to avoid rate limiting
                                 result
                                 })
      )
  
  close(pb)
  
  # Aggregate results by paper
  summary_df <- results_df %>%
    group_by(filename) %>%
    summarize(
      overall_classification = case_when(
        any(str_detect(classification, "RELEVANT")) ~ "RELEVANT", TRUE ~ "NOT RELEVANT" ),
      confidence = list(classification),
      processed_date = Sys.time()
    )
  
  return(list(
    detailed = results_df,
    summary = summary_df
  ))
}

